### PR TITLE
Fix device size dimensions not being correct

### PR DESCRIPTION
**Pull request by trevor-e**

### CODE REVIEW

File: snapshots/snapshots/src/main/kotlin/com/emergetools/snapshots/compose/ComposeSnapshotter.kt [hunk 1]
```diff
@@ -106,10 +106,9 @@ private fun measureComposableSize(
 ): IntSize {
   // Default to 0 if not set which will allow parent to impose constraints on child when
   // AT_MOST set.
-  val heightDpPx: Int =
-    (previewConfig.heightDp ?: 0) * view.resources.displayMetrics.density.toInt()
-  val widthDpPx: Int =
-    (previewConfig.widthDp ?: 0) * view.resources.displayMetrics.density.toInt()
+  val density = view.resources.displayMetrics.density
+  val heightPx = (previewConfig.heightDp ?: 0) * density
```
Comment (trevor-e) [inline snapshots/snapshots/src/main/kotlin/com/emergetools/snapshots/compose/ComposeSnapshotter.kt:110]: The `toInt()` was always rounding down.

File: snapshots/snapshots/src/main/kotlin/com/emergetools/snapshots/compose/DeviceUtils.kt [hunk 2]
```diff
@@ -22,12 +23,13 @@ data class DeviceSpec(
   val dimensionSpec: DimensionSpec
     get() {
       val proportion = densityPpi / MDPI_THRESHOLD
-      val widthDp = widthPixels * proportion
-      val heightDp = heightPixels * proportion
+      val widthDp = widthPixels / proportion
```
Comment (trevor-e) [inline snapshots/snapshots/src/main/kotlin/com/emergetools/snapshots/compose/DeviceUtils.kt:26]: This was another bug.

File: snapshots/snapshots/src/main/kotlin/com/emergetools/snapshots/compose/DeviceUtils.kt [hunk 3]
```diff
@@ -9,7 +9,8 @@ import kotlin.math.abs
 data class DimensionSpec(
   val widthDp: Int? = null,
   val heightDp: Int? = null,
-  val density: Float? = null,
```
Comment (trevor-e) [inline snapshots/snapshots/src/main/kotlin/com/emergetools/snapshots/compose/DeviceUtils.kt:12]: I thought `density` was a horribly overloaded term but can revert if you want.

File: snapshots/snapshots/src/main/kotlin/com/emergetools/snapshots/compose/ComposeSnapshotter.kt [hunk 4]
```diff
@@ -67,7 +67,7 @@ fun snapshotComposable(
     composeView.post {
       // Measure the composable agnostic of the parent constraints to layout properly in activity
       val composableSize = measureComposableSize(composeView, previewConfig)
-      val bitmap = if (previewConfig.showSystemUi == true) {
+      val bitmap = if (previewConfig.showSystemUi == true && previewConfig.device == null) {
```
Comment (trevor-e) [inline snapshots/snapshots/src/main/kotlin/com/emergetools/snapshots/compose/ComposeSnapshotter.kt:70]: This was causing us to always use the emulator size for device snapshots.

File: snapshots/snapshots/src/main/kotlin/com/emergetools/snapshots/compose/ComposeSnapshotter.kt [hunk 5]
```diff
@@ -120,8 +143,8 @@ private fun measureComposableSize(
     previewConfig.heightDp?.let { View.MeasureSpec.UNSPECIFIED } ?: View.MeasureSpec.AT_MOST
 
   view.measure(
-    View.MeasureSpec.makeMeasureSpec(max(view.width, widthDpPx), widthMeasureSpec),
-    View.MeasureSpec.makeMeasureSpec(max(view.height, heightDpPx), heightMeasureSpec),
+    View.MeasureSpec.makeMeasureSpec(max(view.width, widthPx.toInt()), widthMeasureSpec),
```
Comment (trevor-e) [inline snapshots/snapshots/src/main/kotlin/com/emergetools/snapshots/compose/ComposeSnapshotter.kt:146]: I moved the integer rounding to after the px have been calculated. We were losing precision on the density.

File: gradle-plugin/plugin/src/main/kotlin/com/emergetools/android/gradle/tasks/snapshots/LocalSnapshots.kt [hunk 6]
```diff
@@ -147,6 +149,10 @@ abstract class LocalSnapshots : DefaultTask() {
       val output = shell(instrumentationArgs)
       logger.lifecycle(output)
 
+      logger.lifecycle("Dumping logcat logs to file")
```
Comment (rbro112) [inline gradle-plugin/plugin/src/main/kotlin/com/emergetools/android/gradle/tasks/snapshots/LocalSnapshots.kt:152]: This intentional to leave in? Seems maybe this should be a debug setting controlled by an arg.

**Review by rbro112 — APPROVED** [review]
Pre-stamping pending adding a device variant with a width or height override to confirm it matches AS behavior.

**Pull request merged**